### PR TITLE
Add support for new auction entrypoints

### PR DESCRIPTION
**Pull request by wojcik91**
- fix `reserved_slots` handling in `add_bid`
- add new subcommands for `put-transaction` and `make-transaction`:
  - `change_bid_public_key`
  - 'add_reservations`
  - `cancel_reservations` 

Closes https://github.com/casper-ecosystem/casper-client-rs/issues/198

### CODE REVIEW

**Comment by devendran-m** [comment]
Also closes #203

File: public_key.pem
Comment (zajko) [inline public_key.pem:1]: Were these committed intentionally?

File: Cargo.toml [hunk 1]
```diff
@@ -57,6 +57,7 @@ uint = "0.9.5"
 
 [dev-dependencies]
 tempfile = "3.8.1"
+casper-types = { version = "5.0.0", features = ["std", "json-schema", "testing"] }
```
Comment (zajko) [inline Cargo.toml:60]: Shouldn't we be pointing to `feat-2.0` ?

File: public_key.pem
Comment (wojcik91) [reply public_key.pem:1]: Nope. Sorry about that

File: Cargo.toml (hunk 1 above)
Comment (wojcik91) [reply Cargo.toml:60]: Yes, it's configured below.

**Review by zajko — APPROVED** [review]

**Pull request merged**